aws/provider_test.go

```diff
@@ -981,6 +982,55 @@ func testAccPreCheckSkipError(err error) bool {
 	return false
 }
 
+// testSweepOrchestrator is a convenience function that calls testSweepOrchestratorWithData
+//nolint
+func testSweepOrchestrator(jobs []string, r *schema.Resource, region string) error {
+	return testSweepOrchestratorWithData(jobs, r, r.Data(nil), region)
+}
+
+// testSweepOrchestratorWithData launches goroutines to perform sweeping work
+func testSweepOrchestratorWithData(jobs []string, r *schema.Resource, d *schema.ResourceData, region string) error {
+	var wg sync.WaitGroup
+	wgDone := make(chan bool)
+	errChan := make(chan error, len(jobs))
+
+	for _, jobID := range jobs {
+		wg.Add(1)
+
+		go func(id string) {
+			defer wg.Done()
+
+			client, err := sharedClientForRegion(region)
+			if err != nil {
+				errChan <- fmt.Errorf("error getting client: %s", err)
+				return
+			}
+
+			d.SetId(id)
+			err = r.Delete(d, client)
+			if err != nil {
+				errChan <- err
+			}
+		}(jobID)
+	}
+
+	go func() {
+		wg.Wait()
+		close(wgDone)
+		close(errChan)
+	}()
+
+	var errors error
+	var mutex = &sync.Mutex{}
+	for err := range errChan {
+		mutex.Lock()
+		errors = multierror.Append(errors, err)
+		mutex.Unlock()
+	}
```

Comment:
It might be useful to add a comment how this works here, so that future readers of the code know right away. I was initially going to comment that it seemed overly complicated, but then finally figured out that closing `errChan` in the goroutine automatically terminates the iteration in `range errChan`.

Reply:
Done.